site/content/contribute/getting-started/code-review.md

```diff
@@ -38,47 +38,65 @@ If you are a core committer seeking a review
-    * Don't be afraid to pick someone who gives "hard" reviews. Code review feedback is never a personal attack: it should "sharpen" the skills of both the author and the reviewers, not to mention improving the quality of the product.
-    * If you are Mattermost staff, try to take into account the timeoff calendar.
-    * Try to avoid assigning the same person to all of your reviews unless they are related.
-    * When in doubt, ask for recommendations on our community server.
-4. Assign a [QA tester](/contribute/getting-started/core-committers/#qa-testers) and label your pull request with `2: QA Review`.
-    * It is the PM or QA tester's responsibility to determine the scope of required testing, if any.
-    * Reviews by QA may occur at the same time as review by core committers. Be sure to ask for a second review as needed when changes are made.
-    * The QA tester should remove the `2: QA Review` label and add the `QA Review Done` label when their review is done, or if a review is deemed not necessary.
-5. Apply additional labels as necessary:
+2. Immediately assign `1: PM Review`, `2: Dev Review`, and `3: QA Review`.
+    * Your pull request should not be merged until these labels are later removed.
+3. Apply additional labels as necessary:
     * `CherryPick/Approved`: Apply this if the pull request is meant for a quality or patch release.
     * `Do Not Merge/Awaiting PR`: Apply this if the pull request depends on another (e.g. server changes)
     * `Setup Test Server`: Apply this to create a test server with your changes for review.
-    * See [labels](/contribute/getting-started/labels) for more details.
-6. Assign a milestone as necessary.
+    * See [labels](/contribute/getting-started/labels) for additional labels.
+4. Assign a milestone as necessary.
     * Most issues are targeted for an upcoming release, and should be assigned the corresponding milestone.
     * The milestone is mandatory for bug fixes that must be cherry-picked.
-7. Wait for a review.
+5. Request a review from a [product manager](/contribute/getting-started/core-committers/#product-managers).
+    * The choice of product manager is up to you.
+        - In most cases, choose the product mananger embedded with your team.
+        - If your change primarily touches another team's codebase, consider their product mananger.
+        - If in doubt, assign any product mananger and comment about the uncertainty. The product mananger may reassign as appropriate.
+    * [Wait](#if-you-are-awaiting-a-review) for their review to complete before continuing so as to avoid churn if changes are requested.
+    * Remove the `1: PM Review` label only when their review is done.
+        - Product managers ensure the changes meet [user experience guidelines](https://docs.mattermost.com/developer/fx-guidelines.html).
+    * If your changes do not affect the user experience, you may remove `1: PM Review` immediately.
```

Comment:
I have the cut feel that 95% of all PR's are merged without PM Review. And this doesn't seam to be an issue. What is the reasoning on adding `1: PM Review` by default, if it gets removed immediately most of the cases?

Reply:
I have the same gut feel. @wiersgallak, we've long since pretended that `1:PM Review` preceded `2:Dev Review`, but that rarely /seems/ to happen. With these changes, it's obviously a little more painful to pretend.

Thoughts on the future of `1:PM Review`? Should it start as a blocking item as proposed, with it being removed in many cases right away? Does this help improve visibility into the need for PM Review on certain issues (as we hope these changes continue to do for QA Review)?

Reply:
I think it depends on the change.  When there is a new feature and or a change that changes the user experience (text change, new feature) than the PM should be reviewing from the customers perspective  - does this meet the requirements, do what it was intended to do, make the purpose/instruction clear and easy to follow.  

I do like visibility on the changes even if I don't have to explicitly approve them, especially when the change is focused on the backend, the Dev Review is likely to proceed without input from me required. In backend changes, I do not want to be a blocker before Dev reviews get started, because I actually get value from seeing the developers' comments.

Reply:
Let's leave the `1: PM Review` unchanged for now.